docusaurus/docs/operate/configs/relayminer_config.md

```diff
@@ -173,6 +174,21 @@ pprof:
 
 You can learn how to use that endpoint on the [Performance Troubleshooting](../../develop/developer_guide/performance_troubleshooting.md) page.
 
+### `ping`
+
+Configures a `ping` server to test the connectivity of every backend URLs. If
+all the backend URLs are reachable, the endpoint returns a 200 HTTP
+Code. Otherwise, if one or more backend URLs aren't reachable, the service
+returns an 500 HTTP Internal server error.
```

Comment:
Indeed, I think those status codes are more appropriate.
I am on it 👍🏾

Reply:
I'm not so sure about `503 Service Unavailable` in the end. The Relayminer server was ready to handle the request, but subsequent calls are not possible. Therefore, there is an internal error in my opinion.

Reply:
Could you elaborate  on "The Relayminer server was ready to handle the request, subsequent calls are not possible" I'm not sure I follow?

Do you think [504 Gateway Timeout](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/504) is more appropriate?

My thinking is that 500 is quite generic and we should consider using a more specific status if one applies.

Reply:
> My thinking is that 500 is quite generic and we should consider using a more specific status if one applies.

I agree. 500 can be generic, and we must use a more specific status code. I also think we shouldn't use 500 because that implies that the server (in this case, `relayminer`) is completely unusable. But that isn't entirely the case. AFAIK if one supplier isn't reachable anymore, the other configured ones can still function, and the relayminer can continue to mine requests for them.

> Could you elaborate on "The Relayminer server was ready to handle the request, subsequent calls are not possible" I'm not sure I follow?

In my mind, there are three actors here:
1. The client to the `/ping` endpoint. (could be `curl -x GET <addr>:<port>/ping`)
2. The server (`relayminer`)
3. The third-party service (`supplier`)

What I meant by _"The Relayminer server was ready/able to handle requests"_ is the interaction between 1 & 2. And what I meant by the _"subsequent calls are not possible"_ is the interaction between 2 & 3. I hope that makes sense.

My problem with `503 Service Unavailable` is that it implies a temporary state. But we are not so sure about that. The supplier might never come back online.
So yes, I think `504 Gateway Timeout` is more appropriate if the context timeout. But if the the third-party (`supplier`) return an unexpected response, we must use `503 Bad Gateway` IMO. wdyt?